inotify_test.go

```diff
@@ -550,3 +552,40 @@ func TestInotifyDeleteOpenedFile(t *testing.T) {
 	fd.Close()
 	checkEvent(Remove)
 }
+
+func TestINotifyNoBlockingSyscalls(t *testing.T) {
+	getThreads := func() int {
+		cmd := fmt.Sprintf("ls /proc/%d/task | wc -l", os.Getpid())
+		output, err := exec.Command("/bin/bash", "-c", cmd).Output()
+		if err != nil {
+			t.Fatalf("Failed to execute command to check number of threads, err %s", err)
+		}
+
+		n, err := strconv.ParseInt(strings.Trim(string(output), "\n"), 10, 64)
+		if err != nil {
+			t.Fatalf("Failed to parse output as int, err: %s", err)
+		}
+		return int(n)
```

Comment:
I think this should be identical, and should be more portable; even on Linux /bin/bash isn't 100% guaranteed. For example my Alpine system doesn't have it.

    getThreads := func() int {
        d := fmt.Sprintf("/proc/%d/task", os.Getpid())
        ls, err := os.ReadDir(d)
        if err != nil {
            t.Fatalf("reading %q: %s", d, err)
        }
        return len(ls)
    }

It will also makes things a bit faster too as it doesn't spawn a whole bunch of processes.

Reply:
Updated it and merged; thanks!